manager/apis/app/v1alpha1/fybrikapplication_types.go

```diff
@@ -155,25 +155,34 @@ type DatasetDetails struct {
 	Details serde.Arbitrary `json:"details,omitempty"`
 }
 
-// FybrikApplicationStatus defines the observed state of FybrikApplication.
-type FybrikApplicationStatus struct {
+// AssetState defines the observed state of an asset
+type AssetState struct {
+	// Conditions indicate the asset state (Ready, Deny, Error)
+	// +optional
+	Conditions []Condition `json:"conditions,omitempty"`
 
-	// Ready is true if a blueprint has been successfully orchestrated
-	Ready bool `json:"ready,omitempty"`
+	// CatalogedAsset provides a new asset identifier after being registered in the enterprise catalog
+	// +optional
+	CatalogedAsset string `json:"catalogedAsset,omitempty"`
 
-	// Conditions represent the possible error and failure conditions
+	// Endpoint provides the endpoint spec from which the asset will be served to the application
 	// +optional
-	Conditions []Condition `json:"conditions,omitempty"`
+	Endpoint EndpointSpec `json:"endpoint,omitempty"`
+}
 
-	// DataAccessInstructions indicate how the data user or his application may access the data.
-	// Instructions are available upon successful orchestration.
+// FybrikApplicationStatus defines the observed state of FybrikApplication.
+type FybrikApplicationStatus struct {
+	// Ready is true if all specified assets are either ready to be used or are denied access.
 	// +optional
-	DataAccessInstructions string `json:"dataAccessInstructions,omitempty"`
+	Ready bool `json:"ready,omitempty"`
 
-	// CatalogedAssets provide the new asset identifiers after being registered in the enterprise catalog
-	// It maps the original asset id to the cataloged asset id.
+	// ErrorMessage indicates that an error has happened during the reconcile, unrelated to a specific asset
 	// +optional
-	CatalogedAssets map[string]string `json:"catalogedAssets,omitempty"`
+	ErrorMessage string `json:"errorMessage,omitempty"`
```

Comment:
If there are multiple errors are those appended into the same string or will only one of the errors be displayed?